### DISCUSSION TITLE
Excessive CSSStyleSheet instances leaking

### DISCUSSION THREAD

**Discussion by kruart** [chakra-ui/chakra-ui#8706] [General]
### Description

While dynamically updating styles to position elements, new `CSSStyleSheet` instances are created  for each update. This behavior results in a large number of `CSSStyleSheet` instances that cannot be evicted by garbage collector. That can leading to performance degradation and increased memory usage over time.

I have prepared two examples, `Tooltip1` and `Tooltip2`. `Tooltip1` demonstrates the described issue, while `Tooltip2` works as expected. I'm not sure why `DOM Nodes` leak in the first example; it appears to be a bug. The only difference that  `top` and `left` props were moved inside `style` property.

### Link to Reproduction

https://codesandbox.io/p/sandbox/dom-nodes-leak-ff4vsr

### Steps to reproduce

1. It's easier to run the example locally in the `Chrome` browser to reproduce; I'm not sure how to open the performance monitor in the CodeSandbox browser emulator.
3. Run `npm install && npm run start`
4. Open `DevTools` and go to the `Memory` tab.
5. Also open `Performance Monitor` in `DevTools`.
6. Start moving the cursor around the page. You can notice how the number of `DOM Nodes` increases in the `Performance Monitor` window. Those nodes cannot be removed by GC. You can try to press `Collect Garbage` button, it won't help. You can easily move that value to 100K.
7. You could take a heap snapshot in the `Memory` tab to investigate deeper.
8. Then, switch from the `Tooltip1` to the `Tooltip2` example in the `index.tsx` file and try again. It stops leaking.

![performance_monitor_screenshot](https://github.com/user-attachments/assets/6863522f-35a5-4c4e-a8fd-194af638a226)

### Chakra UI Version

2.8.2

### Browser

Google Chrome 127.0.6533.72

### Operating System

- [x] macOS
- [ ] Windows
- [x] Linux

### Additional Information

I didn't test on Windows. But on Mac and Linux it's reproducible.

**Comment by segunadebayo** [comment]
That leak is in v2 Emotion. v2 is in maintenance, so the path forward is v3.
